serviceRegistry.go

```diff
@@ -186,35 +188,64 @@ func (r *ServiceRegistry) redefineCategoryList() {
 			continue
 		}
 		name := filepath.Base(categoryNode.Key)
-		periodResp, err := r.etcd.Get(context.Background(), categoryNode.Key+periodKeySuffix, &client.GetOptions{Sort: true})
-		period := defaultDuration
+
+		//Period
+		period := r.catPeriod(categoryNode.Key)
+
+		//Resilient
+		resilient := r.catResilient(categoryNode.Key)
+
+		//Enabled
+		enabled := r.catEnabled(categoryNode.Key)
+
+		categories[name] = Category{Name: name, Period: period, IsResilient: resilient, Enabled: enabled}
+	}
+	r.categories = categories
+	infoLogger.Printf("%v", r.categories)
+}
+
+func (r *ServiceRegistry) catPeriod(catKey string) (period time.Duration) {
+	period = defaultDuration
+	periodResp, err := r.etcd.Get(context.Background(), catKey+periodKeySuffix, &client.GetOptions{Sort: true})
+	if err != nil {
+		warnLogger.Printf("Failed to get health check period from %v: %v. Using default %v", catKey, err.Error(), defaultDuration)
+	} else {
+		periodInt, err := strconv.Atoi(periodResp.Node.Value)
 		if err != nil {
-			warnLogger.Printf("Failed to get health check period from %v: %v. Using default %v", categoryNode.Key, err.Error(), defaultDuration)
-		} else {
-			periodInt, err := strconv.Atoi(periodResp.Node.Value)
-			if err != nil {
-				warnLogger.Printf("Error reading health check period value '%v'. Using default %v", periodResp.Node.Value, defaultDuration)
-				periodInt = int(defaultDuration.Seconds())
-			}
-			period = time.Duration(periodInt) * time.Second
+			warnLogger.Printf("Error reading health check period value '%v'. Using default %v", periodResp.Node.Value, defaultDuration)
+			periodInt = int(defaultDuration.Seconds())
 		}
-		resilientResp, err := r.etcd.Get(context.Background(), categoryNode.Key+resilientSuffix, nil)
-		resilient := false
+		period = time.Duration(periodInt) * time.Second
+	}
+	return
+}
+
+func (r *ServiceRegistry) catResilient(catKey string) (resilient bool) {
+	resilient = false
+	resilientResp, err := r.etcd.Get(context.Background(), catKey+resilientSuffix, nil)
+	if err != nil {
```

Comment:
Consider returning after `err != nil` so the code which treats the successful case isn't nested.
Same for the `catPeriod` and `catEnabled`.